src/SerializeAST.cc

```diff
@@ -754,10 +774,45 @@ bool IsElementResourceGroup(const Element& element)
     return element.element == Element::CategoryElement && element.category == Element::ResourceGroupCategory;
 }
 
+
+#if _WITH_REFRACT_
+typedef std::vector<const snowcrash::DataStructure*> DataStructures;
+
+void findNamedDataStructures(const snowcrash::Elements& elements, DataStructures& found) {
+    for (snowcrash::Elements::const_iterator i = elements.begin()
+       ; i != elements.end()
+       ; ++i) {
+
+        if (i->element == snowcrash::Element::DataStructureElement) {
+            found.push_back(&(i->content.dataStructure));
+        }
+        else if (!i->content.resource.attributes.empty()) {
+            found.push_back(&i->content.resource.attributes);
```

Comment:
But, yes it looks it is wrong, first example is expanded as "resource".
```
  "content": [
    {
      "element": "category",
      "content": [
        {
          "element": "resource",
          "name": "R",
          "description": "",
          "uriTemplate": "/resource",
          "model": {},
          "parameters": [],
          "actions": [],
          "content": [
            {
              "element": "object",
              "meta": {
                "id": "R"
              },
              "content": [
                {
                  "element": "member",
                  "content": {
                    "key": {
                      "element": "string",
                      "content": "p"
                    },
                    "value": {
                      "element": "string"
                    }
                  }
                }
              ]
            }
          ]
        }
      ]
    },
```

good catch